src/Core/src/Credentials/InstanceProvider.php

```diff
@@ -33,18 +33,18 @@ final class InstanceProvider implements CredentialProvider
 
     private $timeout;
 
-    public function __construct(?HttpClientInterface $httpClient = null, ?LoggerInterface $logger = null, float $timeout = 1.0)
```

Comment:
We cannot remove this because it will break BC.  I suggest leaving this class unchanged and we can deal with the BC layer in the next PR.

Reply:
Got it. Is it Ok if I leave the part that fetched the timeout from env but dont use it?
Also, that test is really awesome i just understood more about how is structured this project by having to debug that error :)

Reply:
Sure. 

For the record: To the next PR that handles the BC layer we need to make the `private $timeout` to support both null and a float. We could remove the constructors third argument and then use `func_get_args()` to see if a third argument was passed to the constructor. If it was we populate our property (and trigger a deprcation), if it wasn't we leave the property to null. 

Then when we need the timeout we use our property and fallback to the environment variable.

Reply:
I suggest to update this class in antoher PR (I can handle it if you want)

Reply:
@jderusse that would be nice since i dont fully understant the BC layer